lib/string.fz

```diff
@@ -454,13 +454,63 @@ string ref : has_equality, hasHash string, ordered string, strings is
     pre
```

Comment:
This does only tangentially belong to this PR but I'm wondering if this precondition !s.isEmpty what you would expect when using a split feature? I would expect to get an empty list in return in such cases.

Reply:
Oh, yes. My bad, sorry. :-)